### ISSUE TITLE
[NEXT-1101] Regression using nested routes with params

### ISSUE THREAD

**Issue by tommybarvaag** [vercel/next.js#49193]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:58 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6020
    Binaries:
      Node: 18.15.0
      npm: 9.5.0
      Yarn: 1.22.19
      pnpm: 8.4.0
    Relevant packages:
      next: 13.3.5-canary.8
      eslint-config-next: 13.3.4
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://github.com/tommybarvaag/nested-routes-with-params

### To Reproduce

Everything works as expected locally with `next dev` but crashes when hosted @Vercel.

Click the link `Go to second params page` on the page https://nested-routes-with-params.vercel.app.

This crashes the page. Resulting in a blank render.

If we peek into the console you can see the following error
![image](https://user-images.githubusercontent.com/24416970/236145305-41a3d758-d2f6-4dc2-bfe7-9f069a7691a5.png)

On further inspection in Vercel logs we can read the following

```
TypeError: Cannot read properties of undefined (reading 'first')
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/app-render/app-render.js:174:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/app-render/app-render.js:450:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/app-render/app-render.js:441:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/app-render/app-render.js:477:75)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/app-render/app-render.js:441:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/app-render/app-render.js:759:57)
```

And

```
TypeError: Cannot read properties of null (reading 'default')
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/build/webpack/loaders/next-edge-ssr-loader/render.js:65:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/web-server.js:349:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/base-server.js:1413:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/base-server.js:1311:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/base-server.js:563:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/web-server.js:258:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/router.js:294:0)
    at (node_modules/.pnpm/next@13.3.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/esm/server/base-server.js:537:0)
```

And

```
 Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error. {
  digest: '951169804'
}
```

### Describe the Bug

Next crashes with nesting routes with params.

The reproduction repo contains the following route structure

```txt
├── app
│   ├── layout.tsx
│   └── page.tsx
│   └── first
│   │   ├── [first]
│   │   │   ├── page.tsx
│   │   │   └── second
│   │   │   │   ├── [second]
│   │   │   │   │   ├── page.tsx
```

When navigating to the [second] page Next will throw an error because the first parameter is undefined.

### Expected Behavior

Shouldn't crash.

### How are you deploying your application? (if relevant)

Vercel

<sub>[NEXT-1101](https://linear.app/vercel/issue/NEXT-1101/regression-using-nested-routes-with-params)</sub>

**Comment by hansottowirtz** [comment]
We're also seeing this error since upgrading from `13.3.1-canary.4` to higher.

Seems like params is not defined here:
https://github.com/vercel/next.js/blob/25a9547cad82e2aea730ce646388786a0d4a0703/packages/next/src/server/app-render/app-render.tsx#L285

**Comment by tommybarvaag** [comment]
Tested with latest canary `13.4.1-canary.1` and this still fails @Vercel but works locally.

```
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:58 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6020
    Binaries:
      Node: 18.15.0
      npm: 9.5.0
      Yarn: 1.22.19
      pnpm: 8.4.0
    Relevant packages:
      next: 13.4.1-canary.1
      eslint-config-next: 13.4.0
      react: 18.2.0
      react-dom: 18.2.0
```

**Comment by huozhi** [comment]
Does latest version 13.4.1 fix your issue?

**Comment by hansottowirtz** [comment]
@huozhi No, still the same error.

**Comment by tommybarvaag** [comment]
@huozhi Upgraded the reproduction site to `13.4.1` and it still gives the same error.

https://nested-routes-with-params.vercel.app

**Comment by huozhi** [comment]
We can repro now, I've marked it as a bug and we're investigating now 🙏

**Comment by tommybarvaag** [comment]
@huozhi I updated the repro to `13.4.2-canary.2` and the page with nested params is now working.

**Comment by timneutkens** [comment]
Should be fixed by #49315 indeed 👍 Thanks for checking!
